tests/Tokenizer/TokensTest.php

```diff
@@ -29,6 +29,587 @@
+;
+  # 1.2',
+            ],
+            'd' => [
+                4,
+                '<?php
+                /* 2 */  '.'
+                ',
+                '<?php
+                /* 2 */  ;
+                ',
+            ],
+            'e' => [
+                3,
+                '<?php # 3
+    ',
+                '<?php # 3
+    '.'
+                ;',
+            ],
+            '4/' => [
+                2,
+                '<?php            ',
+                '<?php            ;',
+            ],
+            '5/' => [
+                1,
+                '<?php ',
+                "<?php\n;",
+            ],
+            // --- before and after
+            [
+                3,
+                '<?php
+# A7
+# B7
+',
+                '<?php
+# A7
+;
+# B7
+',
+            ],
+            [
+                4,
+                '<?php
+                # A9
+                # B
+                ',
+                '<?php
+                # A9
+                ;        '.'
+                # B
+                ',
+            ],
+            // --- after token
+            '10/' => [
+                1,
+                '<?php ',
+                '<?php
+;     ',
+            ],
+            '11/' => [
+                2,
+                '<?php
+',
+                '<?php
+
+;     ',
+            ],
+            '12/' => [
+                1,
+                '<?php ',
+                '<?php
+;',
+            ],
+
+            '14/' => [
+                3,
+                '<?php
+# a',
+                '<?php
+# a
+;     ',
+            ],
+            '16/' => [
+                1,
+                '<?php
+ # 1
+',
+                '<?php
+; # 1
+',
+            ],
+            '17/a' => [
+                3,
+                '<?php
+# 17a
+ # 18a
+',
+                '<?php
+# 17a
+; # 18a
+',
+            ],
+            '17/b' => [
+                3,
+                '<?php
+# 17b
+  # 18b
+',
+                '<?php
+# 17b
+ ; # 18b
+',
+            ],
+            '17/c' => [
+                3,
+                '<?php
+# 17c
+ # 18c
+',
+                '<?php
+# 17c
+ ;# 18c
+',
+            ],
+            '17/d' => [
+                2,
+                '<?php
+/* 17d*/# 18d
+',
+                '<?php
+/* 17d*/;# 18d
+',
+            ],
+            // --- special cases
+            '18/' => [
+                0,
+                '',
+                '<?php ',
+            ],
+            '20/' => [
+                1,
+                '<?php ',
+                '<?php  ',
+            ],
+            '21/' => [
+                1,
+                '<?php ',
+                "<?php\n ",
+            ],
+            '22/' => [
+                2,
+                '<?php
+ '.'
+  '.'
+   '.'
+    '.'
+     '.'
+     '.'
+    '.'
+   '.'
+  '.'
+ ',
+                '<?php
+ '.'
+  '.'
+   '.'
+    '.'
+     '.'
+    # delete this line only 22
+     '.'
+    '.'
+   '.'
+  '.'
+ ',
+            ],
+        ];
```

Comment:
Can you add this case:
```php
    [
        1,
        '<?php
',
        '<?php
TRY_TO_REMOVE_LAST_ELEMENT',
    ],
```
?

Reply:
this: '<?php
TRY_TO_REMOVE_LAST_ELEMENT' does not lint.

However I added:
```
            [
                1,
                '<?php ',
                '<?php
;',
            ],
```

Reply:
when removing the token `;` (@ index 1)  the result is; `<?php\n` , so now there is an empty line left after the removal (before there wasn't) so the function will remove the line,
however `<?php` has a different meaning than `<?php\n`, therefore it is changed to `<?php ` which has the same meaning (https://3v4l.org/VJKHA)

Reply:
Oh, I thought `<?php` would result with syntax error or at least `T_INLINE_HTML` - not with pair of tokens.

Anyway - this would make conflict with `no_trailing_whitespace`.

Reply:
I think there is no conflict (on this level) as `no_trailing_whitespace`  will not trim ['<?php ']( https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.12/tests/Fixer/Whitespace/NoTrailingWhitespaceFixerTest.php#L120) (as it would change the meaning of the token)